app/controllers/submissions_controller.rb

```diff
@@ -48,4 +60,13 @@ def material_submission_params
     )
   end
 
+  def set_contact
+    @contact = Contact.find_by_email(material_submission_params[:contact])
+    if @contact.nil?
+      @contact = Contact.create(:email => material_submission_params[:contact])
+    end
+  end
```

Comment:
You can shorten this with find_or_create_by i.e.

`@contact = Contact.find_or_create_by(email: material_submission_params[:contact])`